_code/InSituTree/vignettes/InSituTree_basicVignette.Rmd

```diff
@@ -0,0 +1,329 @@
+  diff(range(ioprofiles[gene, ])) / max(ioprofiles[gene, ])
+})
+percs
+
+# The 50% quantile is:
+quantile(percs, 0.5)
+
+# Genes satisfying the expression criteria for percentage difference,
+# ie.e genes with expression difference bigger than the 50% of all genes'
+# percentage expression differences, if only six were considered:
+names(percs)[percs > quantile(percs, quantile_perc_difference, na.rm = TRUE)]
+```
+
+The only gene passing **both** of the above filters is JCHAIN, and so in this
+small example where we're considering all cell types together and only six
+genes, only this gene would be considered as having high differentiating power
+at the given granularity level. Note that when fewer cell types or subtypes are
+considered simultaneously, we expect to see more variability in the diffs and
+percs.
+
+10. `return_summary_annotation`: A logical value indicating whether to return
+the final summary annotation.
+This data frame shows the annotation that has been made for each cell at each
+level of the annotation tree. Recommendation is to keep this at its default
+value of TRUE.
+
+```{r strReturnSummary}
+return_summary_df <- TRUE
+```
+
+## Tree-style cell typing
+
+First, we can print a simple structured list to visualize the cell type
+hierarchy we've laid out with our cell type hierarchy list.
+
+```{r cellTypeStructure, fig.width=6, fig.height=4}
+printTree(celltype_hierarchy_list, prefix = "|-")
+```
+
+Once we've verified that this tree shows the hierarchy and cell type
+relationships that we'd expect, we can run supervised nested cell typing.
+
+```{r runInSituTree}
+res <- runInSituTree(
+  x = counts,
+  neg = negmean,
+  full_profiles = ioprofiles,
+  cth = celltype_hierarchy_list,
+  name_for_new_annotation = annotation_names,
+  cohort = NULL,
+  excluded_genes = genes_to_exclude,
+  quantile_absolute_expression_difference_param = quantile_abs_difference,
+  quantile_percent_expression_difference_param = quantile_perc_difference,
+  return_summary_annotation = return_summary_df
+)
+```
+
+In the messages that print out, we see that a handful of cells are removed from
+cell typing if they have zero counts in any of the gene subsets used for typing.
+
+## Review results
+
+The results from running `InSituTree` contain two main components:
+
+1) Nested list of results from each run of `InSituTypeML`
+
+2) Summary of annotations at each level of the tree
+
+The nested results list contains the typical outputs of `InSituType` for each
+round of cell typing, and one additional metric, `ctsPerCell`, which is the 
+counts of the selected genes for each cell at the given round of subclustering.
+This counts metric could be used to evaluate the confidence of a particular
+round of subclustering, since cell type assignments for low-count cells are
+likely to be less confident. 
+Each level of the results list contains:
+
+- `result`: The `InSituType` results plus counts per cell
```

Comment:
- `result`: The `InSituType` results plus counts per cell for **current** level of cell typing. 
- `name`: The **node** name of current level of cell typing.

Reply:
Modified both